Assets/Scripts/EnemyController.cs

```diff
@@ -106,15 +111,27 @@ public void OnDamageTaken()
         gameObject.GetComponent<Renderer>().material.color = Color.Lerp(lowHealthCol, maxHealthCol, health / maxHealth);
     }
 
-    IEnumerator TriggerDamageEffect()
+    IEnumerator doAttack()
     {
```

Comment:
```suggestion
    IEnumerator Attack()
    {
```
You mean `Attack`? 😅

Reply:
We already have a function called Attack, but I guess it's fine since different types

Reply:
Actually, I cannot name a IEnumerator the same as a void function :D

Reply:
what :D that's not right then. Let's name them in a way where their functionalities are clearly separated cause now I'm confused what is the difference between these two. Maybe there's animation to one of them, so name it `AttackWithAnimation`, or maybe the attack type is different or something else. Please rename them

Reply:
Oh I now checked the code and I think you could rename `Attack` function to `StartAttackProcess` or something similar

Reply:
Cause now also `Attack` and `AttackPlayer` is totally the same for me :D